src/theory/quantifiers/ematching/trigger.cpp

```diff
@@ -859,12 +860,15 @@ Node Trigger::getInversion( Node n, Node x ) {
         }
         x = Rewriter::rewrite( x );
       }else{
-        Assert( cindex==-1 );
+        Assert(!cindexSet);
         cindex = i;
+        cindexSet = true;
       }
     }
-    Assert( cindex!=-1 );
-    return getInversion( n[cindex], x );
+    if (cindexSet)
```

Comment:
The fix has a slightly different semantics than before (previously not setting the cindex was considered an error). Is this on purpose?

Reply:
Yes, this is a case that we should explicitly guard the segfault (it is in code that is very infrequently run).

Reply:
Yes, we should definitely not segfault. What I meant to ask is if we do the right thing if `cindexSet` is false (i.e. if returning a null node is the correct course of action). It does sound like it is though.